### ISSUE TITLE
nilearn>=0.6.0 installs empty `sklearn` dependency

### ISSUE THREAD

**Issue by rmarkello** [nilearn/nilearn#2288]
Hi nilearn team!

I was recently creating a new Python environment and noticed the dependency chain for `nilearn` had changed. In particular, it seems that installing version 0.6.0 of `nilearn` into a clean environment automatically installs the (fake) [`sklearn`](https://pypi.org/project/sklearn/) package (in addition to the actual `scikit-learn` package). 

This dependency is not installed when using the previous release (0.6.0b0). Indeed, it looks like this functionality may have been added with #2201, and I believe the relevant code is here:

https://github.com/nilearn/nilearn/blob/9173aaca590df50bbaa6122bc744291617f36b90/setup.py#L33-L44

Specifically, L43 of that snippet adds the `sklearn` dependency to the `install_requires` list.

I may be missing something (the referenced PR has quite a lot happening so it was tough to follow everything!), but given that `sklearn` seems to only exist to prevent malicious actors (the package doesn't actually do anything...), would it be worth removing as a dependency?

**Comment by GaelVaroquaux** [comment]
Right! We should have a dependency on "scikit-learn", not "sklearn". Good
catch!

**Comment by rmarkello** [comment]
You do have a dependency on `scikit-learn` (L41 in the code snippet above makes sure of that)! However, I think there _is_ a reason that `sklearn` is listed in the `nilearn/version.py` file given some of the commits in #2201(e.g., 4cfb901), but I just can't tell post-hoc what that rationale is...

Nonetheless, I can't think of a good reason why it should be required in the final build wheel.

Just in case it helps, the full dependency tree for `nilearn` (from `pip install nilearn`) as of 0.6.0 is:

```
- nilearn
  - joblib [required: >=0.11]
  - nibabel [required: >=2.0.2]
    - numpy [required: >=1.12]
  - numpy [required: >=1.11]
  - scikit-learn [required: >=0.19]
    - joblib [required: >=0.11]
    - numpy [required: >=1.11.0]
    - scipy [required: >=0.17.0]
  - scipy [required: >=0.19]
  - sklearn [required: Any]
```

**Comment by kchawla-pi** [comment]
It was deliberately added because one of our CIs, Appveyor, refused to work without it. 

My assessment was that trying to fix was a lot of cost for little benefit, since it didn't affect the code functionality or quality.
This also opened up the possibility that some systems out in the wild might glitch as well. 

Between choosing that possibility vs a cosmetic difference which will be in play only once for most users, during Nilearn install time, I decided to add the dummy installation back in.

**Comment by kchawla-pi** [comment]
There were also import failures, as noted in the commit history. I don't remember the details of it now,

**Comment by bthirion** [comment]
@kchawla-pi are you taking another look at that ?
Thx

**Comment by kchawla-pi** [comment]
This is not a priority right now.

**Comment by emdupre** [comment]
Should we re-visit this now that we're updated our CI, @kchawla-pi  ? I'll add it to the 0.7.x project board, but feel free to move it to the 0.8.x one instead if you think it will take too long to address !

**Comment by thomasbazeille** [comment]
tackled by the merged PR, thanks
